xdsl/dialects/arith.py

```diff
@@ -153,16 +153,26 @@ def __init__(
     ) -> None: ...
 
     @overload
-    def __init__(self, value: Attribute, value_type: Attribute) -> None: ...
+    def __init__(
+        self,
+        value: Attribute,
+        value_type: Attribute,
+    ) -> None: ...
 
     def __init__(
         self,
-        value: IntegerAttr | FloatAttr[AnyFloat] | Attribute,
+        value: IntegerAttr | FloatAttr[AnyFloat] | DenseIntOrFPElementsAttr | Attribute,
```

Comment:
In my understanding, the latest changes don't correspond to Mathieu's suggestion. Why allow passing `Attribute` here, if it's not allowed by the constraint on `value`? Wouldn't this cover all possible values to be used as constants?
```suggestion
        value: IntegerAttr | FloatAttr[AnyFloat] | DenseIntOrFPElementsAttr,
```
If so, why let users provide the value_type, if it can always be found on the input?

Reply:
It's possible that I also misunderstood, but the type definitely seems overly permissive for this style of init that only lets you assign valid things

Reply:
I think it's clear now